beacon-chain/db/kv/block_roots.go

```diff
@@ -0,0 +1,76 @@
+package kv
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	bolt "go.etcd.io/bbolt"
+	"go.opencensus.io/trace"
+)
+
+// BlockRoot returns the block root from the DB using the state root
+func (s *Store) BlockRoot(ctx context.Context, stateRoot [32]byte) ([32]byte, error) {
+	_, span := trace.StartSpan(ctx, "BeaconDB.BlockRoot")
+	defer span.End()
+
+	var blockRoot [32]byte
+	err := s.db.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(blockRootsBucket)
+		root := bucket.Get(stateRoot[:])
+		if root == nil {
+			return nil
+		}
+		blockRoot = [32]byte(root)
+		return nil
+	})
+
+	return blockRoot, err
+}
+
+// SaveBlockRoot saves state root to block root mapping into DB
+func (s *Store) SaveBlockRoot(ctx context.Context, stateRoot [32]byte, blockRoot [32]byte) error {
```

Comment:
Like i mentioned earlier, we don't need consumers using/accessing these methods. This can come under `SaveBlock`